### PR TITLE
docs: gateway doc changes p2

### PR DESCRIPTION
**Pull request by hamidzr**
## Ticket

https://hpe-aiatscale.atlassian.net/browse/RM-281

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/real_config.yml [hunk 1]
```diff
@@ -1409,9 +1409,8 @@ jobs:
             - model_hub/dist
             - docs/build
             - docs/site
-      - run: tar czf docs.tgz docs/site/html
       - store_artifacts:
-          path: docs.tgz
```
Comment (hamidzr) [inline .circleci/real_config.yml:1414]: shouldn't reach main

File: docs/setup-cluster/k8s/internal-task-gateway.rst [hunk 2]
```diff
@@ -15,13 +15,16 @@ refer to the sections below to see the configuration changes and controller requ
...
+   communication between the external cluster and the main one. This could include setting up a
+   firewall, using a VPN, IP white-listing, K8s Network Policies, native cloud solutions, or other
+   security measures. Some Determined tasks including JupyterLab Notebooks and shells already have
+   secure transport built-in.
 
 Limitations:
 
--  Exposing proxies in multi-slot jobs is not supported. Currently this only includes experiments
-   running distributed training that want to manually expose proxies to the outside world.
+-  Proxying to multi-node trials is not supported: Exposing proxies in multi-slot jobs is not supported.
```
Comment (NicholasBlaskey) [inline docs/setup-cluster/k8s/internal-task-gateway.rst:26]:
```suggestion
-  Proxying to multi-node trials is not supported
```

I think a notebook that requests 2 gpus would count as a "multi-slot" job

we do support that case

File: docs/setup-cluster/k8s/internal-task-gateway.rst [hunk 3]
```diff
@@ -119,11 +122,6 @@ If you're running Determined outside of the K8s cluster, for example on your loc
...
- Release Notes
-###############
-
-TBD
-
--  Mention docs
--  Mention current limitations?
+For allowing Determined tasks to connect to master that's running locally on your machine, you can
+use services like ngrok or a reverse SSH tunnel if you have access to a public IP like so: `ssh -R
+8080:localhost:8080 aws-dev.prv -N -o ServerAliveInterval=60 -o ServerAliveCountMax=10`
```
Comment (NicholasBlaskey) [inline docs/setup-cluster/k8s/internal-task-gateway.rst:127]: nice

File: master/internal/rm/kubernetesrm/gateway_service.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ type gatewayService struct {
 	portRangeEnd       int
 }
 
-// chat: these probably could use a better names or packaging.
+// TODO: these probably could use a better names or packaging.
 type gatewayResourceComm struct {
```
[the remark this answers is not in the document]
Comment (NicholasBlaskey) [reply master/internal/rm/kubernetesrm/gateway_service.go:34]: I think this turned out decently well imo

**Review by NicholasBlaskey — APPROVED** [review]

**Pull request merged**